### PR TITLE
style: introduce aliasing, sort imports

### PR DESCRIPTION
**Pull request by kwajiehao**
# Overview

This PR is the frontend equivalent of PR #[194](https://github.com/isomerpages/isomercms-backend/pull/194) in the backend.  Broadly speaking, we wish to introduce aliases for our import paths, and to sort these imports. 

However, since the frontend uses Create React App, or CRA, with its own configurations behind the scenes it was not as straightforward as compared to the backend.

This PR makes the following changes:
1. Use CRACO to define aliases, instead of `module-alias`
2. Replace import paths with newly-defined aliases
3. Define sorting config with `eslint-simple-import-sort`
4. Autofix import sorts using eslint

The key files to read are: 
- `package.json` - addition of module aliases and jest aliases, replacement of `react-scripts` with `craco` in the `scripts` section
- `jsconfig.json` - addition of VS Code webpack aliases
- `craco.config.json` - addition of webpack aliases to extend default CRA webpack config
- `eslintrc.json` - defining import sorting rules using `eslint-plugin-simple-import-sort`
- `cypress/plugins/index.js` - defining aliases for cypress using `@cypress/webpack-preprocessor`

### 1. Using CRACO to define aliases

Our previous trick of using the `module-alias` package does not work with the frontend, because of the webpack step that is run when compiling the React app (see this [comment](https://github.com/ilearnio/module-alias/issues/37#issuecomment-464212934) from one of the `module-alias` collaborators).

The solution therefore is to define aliases on the webpack level. However, since CRA comes with its own webpack config, we can't just define a `webpack.config.js` file at the repo root. Instead, we need to use the CRACO (Create React App Configuration Override) library (refer to the [webpack api for CRACO](https://github.com/gsoft-inc/craco/blob/master/packages/craco/README.md#webpack-api), and an [example from stack overflow](https://stackoverflow.com/a/65746792)). CRACO helps us to extend the CRA webpack config and defines the aliases for us.

Note however that we still need to install `module-alias` and define aliases in `package.json` for use with eslint. Other steps we've done for the backend, such as including import resolvers for the eslint configs also need to be applied for the frontend (the additional step that was done here is that we included the `extension` option so that aliases can be identified not just for `.js` files but also for `.jsx` and `.scss` files).

### 2. Replacing import paths with aliases

Import paths were manually replaced with their aliases by using the search and replace functionality on the VS Code IDE. This was subsequently tested with our E2E tests.

### 3. Sorting with `eslint-plugin-simple-import-sort`
For the frontend, we'll be using the `eslint-plugin-simple-import-sort`  instead of `eslint-plugin-import`, since it has a simpler API and achieves the same effect. If we wish to achieve more standardization, we can use the `eslint-plugin-import` instead.

We referenced this [file](https://github.com/lydell/eslint-plugin-simple-import-sort/blob/main/examples/.eslintrc.js) when creating our sorting order.

Note that we were unable to use this because we use `require`s instead of `import`s in the backend. If we make the switch to `import`, we can then switch to this plugin for the backend as well.

### 4. Autofix import sorts

This PR also runs the following command to automatically fix import sorting related issues: `npx eslint --ext .js --ext .jsx --plugin eslint-plugin-simple-import-sort . --fix`

After this, I also manually remove comments related to imports, such as `// Import components`, or `// Import utils`, as they are now redundant.

## Aliases for Jest, VS Code, and Cypress

Unfortunately, although we've defined aliases for both eslint as well as CRA (through CRACO), our Jest route selector tests and Cypress E2E tests no longer work, and IntelliSense for VS Code breaks as well. Therefore, we need to define aliases for both Jest and VS Code. 

For VS Code, we also referred to their [documentation for defining webpack aliases](https://code.visualstudio.com/docs/languages/jsconfig#_using-webpack-aliases) and defined the webpack aliases using glob patterns in a `jsconfig.json` file.

For Jest, we took reference from [this article](https://medium.com/@justintulk/solve-module-import-aliasing-for-webpack-jest-and-vscode-74007ce4adc9) and used the Jest `moduleNameMapper` to map the paths to the aliases.

For Cypress, we used the `@cypress/webpack-preprocessor` package to define the same aliases for our E2E tests. Ideally, we would have passed our webpack config into the function provided by this package, but since we are using CRACO, we don't actually have a webpack config to pass in and so we redefined the webpack aliases here. I'm sure there are alternative ways of going about this, and we can explore this when we have more time in the future.

## Other changes made
- Remove unused tree-related files (`tree-builder`, `tree-utils`) and packages (`@atlaskit/tree`)
- Use `prettier` instead of `plugin:prettier/recommended`. See backend PR #[195](https://github.com/isomerpages/isomercms-backend/pull/195) for more details
- Add `no-underscore-dangle` rule

### CODE REVIEW

**Review by prestonlimlianjie — CHANGES_REQUESTED** [review]
lgtm apart from one small issue where Images and Folders tests are being commented out. Is this intentional?

File: src/__tests__/RouteSelector.jsx [hunk 1]
```diff
@@ -22,140 +25,88 @@ const FOLDERS_LAYOUT_TEXT = "Folders layout mock text"
...
+  __esModule: true,
+  default: () => <div>{EDIT_HOMEPAGE_LAYOUT_TEXT}</div>,
+}))
+
+jest.mock("@layouts/Workspace", () => ({
+  __esModule: true,
+  default: () => <div>{WORKSPACE_LAYOUT_TEXT}</div>,
+}))
+
+// jest.mock("@layouts/Images", () => {
```
[the remark this answers is not in the document]
Comment (kwajiehao) [reply src/__tests__/RouteSelector.jsx]: I've uncommented out the tests and filed an issue for these failing Jest mocks #513

**Comment by prestonlimlianjie** [comment]
There are also some styling differences - probably from style import issues.

### Commit `f58cf86` (current `style/aliases` branch)
Some elements are using System-UI font like the repo name (`e2e-test-repo`), text in the Cards (`Homepage`, `Navigation Bar` etc).

Screenshot 2021-06-09 at 12 30 26 PM

### Commit `531e571` (current `develop` branch)
Elements are using Lato font
Screenshot 2021-06-09 at 12 31 40 PM

**Comment by kwajiehao** [comment]
> There are also some styling differences - probably from style import issues.
> 
> ### Commit `f58cf86` (current `style/aliases` branch)
> Some elements are using System-UI font like the repo name (`e2e-test-repo`), text in the Cards (`Homepage`, `Navigation Bar` etc).
> 
> Screenshot 2021-06-09 at 12 30 26 PM
> 
> ### Commit `531e571` (current `develop` branch)
> Elements are using Lato font
> Screenshot 2021-06-09 at 12 31 40 PM

Fixed with [6682d98](https://github.com/isomerpages/isomercms-frontend/pull/511/commits/6682d982f61e51b456863a8e4ead1efa053cfcd2) - read the commit message for more details

**Review by prestonlimlianjie — APPROVED** [review]
lgtm!

**Pull request merged**